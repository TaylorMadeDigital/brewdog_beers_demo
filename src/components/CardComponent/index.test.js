import React from 'react';
import { render, cleanup } from "@testing-library/react";
import { BrowserRouter as Router } from 'react-router-dom';
import { CardComponent } from './index';
import cards from './index.fixtures';

describe('<CardComponent />', () => {

  afterEach(cleanup);

  it('snapshot: renders default state', () => {
    const { asFragment } = render(<Router><CardComponent item={cards[0]} /></Router>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('snapshot: renders without link', () => {
    const { asFragment } = render(<Router><CardComponent item={cards[1]} /></Router>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('snapshot: renders with image', () => {
    const { asFragment } = render(<Router><CardComponent item={cards[2]} /></Router>);
    expect(asFragment()).toMatchSnapshot();
  });
});
