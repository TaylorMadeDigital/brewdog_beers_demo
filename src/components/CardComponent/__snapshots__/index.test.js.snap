// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardComponent /> snapshot: renders default state 1`] = `
<DocumentFragment>
  <article
    data-testid="card"
  >
    <a
      href="/url/to"
    >
      <header
        data-testid="cardHeading"
      >
        <h2
          class=""
          data-testid="heading"
        >
          Sample beer
        </h2>
        <h3
          class=""
          data-testid="heading"
        >
          Beer subtitle
        </h3>
      </header>
    </a>
  </article>
</DocumentFragment>
`;

exports[`<CardComponent /> snapshot: renders with image 1`] = `
<DocumentFragment>
  <article
    data-testid="card"
  >
    <a
      href="/url/to"
    >
      <header
        data-testid="cardHeading"
      >
        <h2
          class=""
          data-testid="heading"
        >
          Sample beer
        </h2>
        <h3
          class=""
          data-testid="heading"
        >
          Beer subtitle
        </h3>
        <img
          alt="Sample beer"
          src="https://via.placeholder.com/300/09f/fff.png"
        />
      </header>
    </a>
  </article>
</DocumentFragment>
`;

exports[`<CardComponent /> snapshot: renders without link 1`] = `
<DocumentFragment>
  <article
    data-testid="card"
  >
    <header
      data-testid="cardHeading"
    >
      <h2
        class=""
        data-testid="heading"
      >
        Sample beer
      </h2>
      <h3
        class=""
        data-testid="heading"
      >
        Beer subtitle
      </h3>
    </header>
  </article>
</DocumentFragment>
`;
